### PR TITLE
ARROW-16240: [Python] Support row_group_size/chunk_size keyword in pq.write_to_dataset with use_legacy_dataset=False

### PR DESCRIPTION
**Pull request by AlenkaF**

### CODE REVIEW

File: python/pyarrow/parquet/__init__.py [hunk 1]
```diff
@@ -3049,6 +3049,7 @@ def file_visitor(written_file):
         # extract non-file format options
         schema = kwargs.pop("schema", None)
         use_threads = kwargs.pop("use_threads", True)
+        chunk_size = kwargs.pop("chunk_size", None)
```
Comment (jorisvandenbossche) [inline python/pyarrow/parquet/__init__.py:3052]: Can you also catch `row_group_size`, as that is the documented keyword in `write_table` (it keeps `chunk_size` as an alias through `row_group_size = kwargs.pop('chunk_size', row_group_size)` inside `write_table`.
Probably giving precedence to `row_group_size` over `chunksize`

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks!

File: python/pyarrow/parquet/__init__.py (hunk 1 above)
Comment (AlenkaF) [reply python/pyarrow/parquet/__init__.py:3052]: Yes, will do!

File: python/pyarrow/parquet/__init__.py [hunk 2]
```diff
@@ -3049,6 +3049,11 @@ def file_visitor(written_file):
...
+        row_group_size = kwargs.pop("row_group_size", None)
+
+        if (row_group_size or chunk_size) is not None:
+            row_group_size = row_group_size if row_group_size else chunk_size
```
Comment (jorisvandenbossche) [inline python/pyarrow/parquet/__init__.py:3056]:
```suggestion
        row_group_size = row_group_size if row_group_size is not None else chunk_size
```

(slight simplification)

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request closed without merging**